internal/v2/database/text_filter.go

```diff
@@ -0,0 +1,132 @@
+package database
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/zitadel/logging"
+)
+
+type TextFilter[T text] struct {
+	Filter[textCompare, T]
+}
+
+func NewTextEqual[T text](t T) *TextFilter[T] {
+	return newTextFilter(textEqual, t)
+}
+
+func NewTextUnequal[T text](t T) *TextFilter[T] {
+	return newTextFilter(textUnequal, t)
+}
+
+func NewTextEqualInsensitive[T text](t T) *TextFilter[T] {
+	return newTextFilter(textEqualInsensitive, t)
+}
+
+func NewTextUnequalInsensitive[T text](t T) *TextFilter[T] {
+	return newTextFilter(textUnequalInsensitive, t)
+}
+
+func NewTextStartsWith[T text](t T) *TextFilter[T] {
+	return newTextFilter(textStartsWith, t)
+}
+
+func NewTextStartsWithInsensitive[T text](t T) *TextFilter[T] {
+	return newTextFilter(textStartsWithInsensitive, t)
+}
+
+func NewTextEndsWith[T text](t T) *TextFilter[T] {
+	return newTextFilter(textEndsWith, t)
+}
+
+func NewTextEndsWithInsensitive[T text](t T) *TextFilter[T] {
+	return newTextFilter(textEndsWithInsensitive, t)
+}
+
+func NewTextContains[T text](t T) *TextFilter[T] {
+	return newTextFilter(textContains, t)
+}
+
+func NewTextContainsInsensitive[T text](t T) *TextFilter[T] {
+	return newTextFilter(textContainsInsensitive, t)
+}
+
+func newTextFilter[T text](comp textCompare, t T) *TextFilter[T] {
+	return &TextFilter[T]{
+		Filter: Filter[textCompare, T]{
+			comp:  comp,
+			value: t,
+		},
+	}
+}
+
+func (f TextFilter[T]) Write(stmt *Statement, columnName string) {
+	if f.comp.isInsensitive() {
+		f.writeCaseInsensitive(stmt, columnName)
+		return
+	}
+	f.Filter.Write(stmt, columnName)
+}
+
+func (f *TextFilter[T]) writeCaseInsensitive(stmt *Statement, columnName string) {
+	stmt.WriteString("LOWER(")
+	stmt.WriteString(columnName)
+	stmt.WriteString(") ")
+	stmt.WriteString(f.comp.String())
```

Comment:
this assumes that the value is always passed lower case, shouldn't we make sure here:
```suggestion
	stmt.WriteString(strings.ToLower(f.comp.String()))
```

Reply:
current tests should be extended to reflect that as well

Reply:
added strings.ToLower to the constructors